qutip/core/coefficient.py

```diff
@@ -53,6 +53,7 @@ def _return(base, **kwargs):
     np.ndarray: InterCoefficient,
     scipy.interpolate.PPoly: InterCoefficient.from_PPoly,
     scipy.interpolate.BSpline: InterCoefficient.from_Bspline,
+    numbers.Number: ConstantCoefficient,
```

Comment:
Should we then also include `float` in the definition of `CoefficientLike`? (Then, in `nm_mcsolve`, we could remove some "`float |`")

Also, the `if isinstance(rate, numbers.Number)` branching in `nm_mcsolve._parse_op_and_rate` shouldn't be necessary any more with this, right?